src/libraries/System.Private.CoreLib/src/System/Globalization/GregorianCalendarHelper.cs

```diff
@@ -480,15 +373,13 @@ public int[] Eras
         {
             get
             {
-                if (m_eras == null)
+                EraInfo[] eraInfo = m_EraInfo;
```

Comment:
>                 [](http://example.com/codeflow?start=0&length=16)

why got rid of m_eras and filling the returning array every time the property get called?

Reply:
Is array cloning significantly cheaper for such small arrays?

Reply:
Array.Clone usually will use Buffer.Memmove for the arrays not having GC pointers (like the case here). I believe that would be faster, but I didn't measure it to give you how much would be faster.

Reply:
A quick benchmark shows that for a 5 element array the cached version is about 10 times slower:
|        Method |     Mean |    Error |   StdDev |
|-------------- |---------:|---------:|---------:|
| GetErasCached | 97.86 ns | 0.999 ns | 0.594 ns |
|       GetEras | 10.93 ns | 1.007 ns | 0.666 ns |

Reply:
Interesting. That is very surprising. Anyway, thanks for measuring it. we should be good then.